src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -228,7 +228,56 @@ private function evaluateBracket(string $expr, mixed $value): array
             return $this->evaluateFilter($innerExpr, $value);
         }
 
-        // quoted strings for object keys
+        // comma-separated values, e.g. `['key1', 'key2', 123]` or `[0, 1, 'key']`
+        if (str_contains($expr, ',')) {
+            $parts = $this->parseCommaSeparatedValues($expr);
+
+            $result = [];
+            $keysIndices = array_keys($value);
+            $isList = array_is_list($value);
+
+            foreach ($parts as $part) {
+                $part = trim($part);
+
+                if (preg_match('/^([\'"])(.*)\1$/', $part, $matches)) {
+                    $key = stripslashes($matches[2]);
```

Comment:
this needs to unescape the string properly to support unicode escape sequences.

Reply:
Indeed, just waiting for #60681 to be merged